wgpu-core/src/device/queue.rs

```diff
@@ -548,6 +553,10 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let pending_writes_cmd_buf = device.pending_writes.command_buffer.as_mut().unwrap();
         let mut trackers = device.trackers.lock();
 
+        let tmp_ref_count = ptr::NonNull::new(Box::into_raw(Box::new(AtomicUsize::new(1))))
```

Comment:
oh no, that's definitely not what we want here :)
Here it's too late to handle the deleted buffer. Instead, we should grab a refcount from it where it's used.
So `buffer_memory_init_actions` needs to carry that refcount to us.